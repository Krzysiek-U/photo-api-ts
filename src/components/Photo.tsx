import { Link } from "react-router-dom";

interface PhotoProps {
  photo: { id: number, author: string };
}

const Photo = ( props: PhotoProps ) => {

  return (    
    <div className='photo-container contrast'>
      <Link to={`/photo/?id=${props.photo.id}&autor=${props.photo.author}`}>

        <img
          alt = { `Zdjęcie o numerze ${props.photo.id}, autor zdjęcia: ${props.photo.author}` }
          className='photo'
          src={ `https://picsum.photos/id/${props.photo.id}/300/200/` }>                      
        </img>
        </Link>
      <h2 className='contrast'>{props.photo.author}</h2>
    </div>
  );
};

export default Photo;