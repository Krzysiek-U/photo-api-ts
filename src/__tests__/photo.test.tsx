import React from "react";
import { BrowserRouter as Router} from "react-router-dom";
import { render } from '@testing-library/react';
import Photo from '../components/Photo';

const photo = { id : 1, author : 'Alejandro Escamilla' };
const index = 1;

describe( 'Photo component', () => {
    test('should match the snapshot', () => {
        const { container } = render(<Router>  <Photo key={index} photo={photo} /> </Router>  );
        expect(container).toMatchSnapshot();
        //expect(1).toBe(1);
    })    
});