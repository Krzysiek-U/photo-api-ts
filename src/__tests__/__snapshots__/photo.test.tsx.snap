// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo component should match the snapshot 1`] = `
<div>
  <div
    class="photo-container contrast"
  >
    <a
      class="link photo"
      href="/photo/?id=1&autor=Alejandro Escamilla"
      title="Przejdź do zdjęcia o numerze 1, autor zdjęcia: Alejandro Escamilla"
    >
      <img
        alt="Zdjęcie o numerze 1, autor zdjęcia: Alejandro Escamilla"
        class="photo"
        src="https://picsum.photos/id/1/300/200/"
      />
      <div
        class="contrast photo title"
      >
        Alejandro Escamilla
      </div>
    </a>
  </div>
</div>
`;
